### PR TITLE
TOOLS/PERF: Add a method to reset sn in ucp_put case

### PR DESCRIPTION
**Pull request by huaxiangfan**
## What
Add a method reset_buffers to reset sn for both send/receive buffers in ucp_put cases. This method is then used in run_pingpong and run_stream_uni.

## Why ?
Without the resetting a performance degradation was encountered in run_stream_uni. https://github.com/openucx/ucx/pull/6951 reported this issue and proposed a simple patch. This patch is for the same purpose but takes yosefe's comments in.

### CODE REVIEW

File: src/tools/perf/lib/ucp_tests.cc [hunk 1]
```diff
@@ -504,6 +504,20 @@ class ucp_perf_test_runner {
...
     }
 
+    void reset_buffers(const void *src, size_t offset, size_t count)
+    {
+        if (CMD == UCX_PERF_CMD_PUT) {
+            m_perf.allocator->memcpy((uint8_t *)m_perf.recv_buffer + offset,
+                                     m_perf.allocator->mem_type,
+                                     src, UCS_MEMORY_TYPE_HOST,
+                                     count);
+            m_perf.allocator->memcpy((uint8_t *)m_perf.send_buffer + offset,
```
Comment (yosefe) [inline src/tools/perf/lib/ucp_tests.cc:514]: can you pls use UCS_PTR_BYTE_OFFSET(m_perf.recv_buffer, offset)
same for sender

File: src/tools/perf/lib/ucp_tests.cc [hunk 2]
```diff
@@ -504,6 +504,20 @@ class ucp_perf_test_runner {
         ucp_worker_fence(m_perf.ucp.worker);
     }
 
+    void reset_buffers(const void *src, size_t offset, size_t count)
```
Comment (yosefe) [inline src/tools/perf/lib/ucp_tests.cc:507]: 1. `psn_t src`
2. no need to pass count - it's always one psn number

**Comment by yosefe** [comment]
ok to test
@huaxiangfan is with NVIDIA

File: src/tools/perf/lib/ucp_tests.cc (hunk 1 above)
Comment (huaxiangfan) [reply src/tools/perf/lib/ucp_tests.cc]: Done with the 2nd commit

File: src/tools/perf/lib/ucp_tests.cc (hunk 2 above)
Comment (huaxiangfan) [reply src/tools/perf/lib/ucp_tests.cc]: Done with the 2nd commit. Thanks.

**Comment by huaxiangfan** [comment]
I observed this when I was running "UCX_NET_DEVICES=mlx5_bond_1:1 UCX_TLS=ud ../bin/ucx_perftest -t ucp_put_bw -s 409600 192.168.30.3" as a perftest client:
Without the patch (i.e. the current situation), we could see perftest server exits when ctrl+c perftest client. Because by default two test rounds are conducted with the first one being warm up. Without resetting the sn as the patch will do, server exits wait_last_iter() immediately after entering the second round. that was wrong and might be the root cause why huge performance degradation was seen.
With the patch applied, we could see perftest server *hang* when ctrl+c perftest client. Because in this case client didn't get a chance to send sn to indicate end. That is an expected behavior in my humble opinion. Or We could enhance it in future. 
Anyway this patch should be irrelevant to this. If we specifiy "-w 0" to cancel warm-up, pertest server would hang when ctrl+c perftest client with or without this patch applied. If we specifiy a value for iteration number by using "-n <number>", pertest server would exit when ctrl+c perftest client with or without this patch applied.

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**